google/cloud/bigquery_storage_v1beta2/writer.py

```diff
@@ -91,10 +91,11 @@ def __init__(
         self._futures_queue = queue.Queue()
         self._inital_request_template = initial_request_template
         self._metadata = metadata
+        self._opening = threading.Lock()
```

Comment:
I'd recommend documenting here (and/or in _open) what the lock is actually guarding.